torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -38,14 +58,71 @@ bool ShardingUtil::SetHloSharding(LoweringContext* lowering_ctx) {
   return is_sharded;
 }
 
+xla::OpSharding ShardingUtil::CreateOpSharding(const py::list& tile_assignment,
+                                               bool replicated, bool manual) {
+  XLA_CHECK(!(replicated && manual))
+      << "Invalid arguments: replicated=" << replicated
+      << ", manual=" << manual;
+
+  xla::OpSharding sharding;
+  if (replicated) {
+    sharding = xla::HloSharding::Replicate().ToProto();
+  } else if (manual) {
+    sharding = xla::HloSharding::Manual().ToProto();
+  } else {
+    // Sharding type is tiled
+    auto dims = TileAssignmentDimensions(tile_assignment);
+    xla::Array<int64_t> tile_array(dims);
+    switch (dims.size()) {
+      case 1:
+        tile_array.Each([&](absl::Span<const int64_t> indices, int64_t* v) {
+          *v = tile_assignment[indices[0]].cast<int64_t>();
+        });
+        break;
+      case 2:
+        tile_array.Each([&](absl::Span<const int64_t> indices, int64_t* v) {
+          auto r = tile_assignment[indices[0]].cast<py::list>();
+          *v = r[indices[1]].cast<int64_t>();
+        });
+        break;
+      case 3:
+        tile_array.Each([&](absl::Span<const int64_t> indices, int64_t* v) {
+          auto r = tile_assignment[indices[0]].cast<py::list>();
+          r = r[indices[1]].cast<py::list>();
+          *v = r[indices[2]].cast<int64_t>();
+        });
+        break;
+      case 4:
+        tile_array.Each([&](absl::Span<const int64_t> indices, int64_t* v) {
+          auto r = tile_assignment[indices[0]].cast<py::list>();
+          r = r[indices[1]].cast<py::list>();
+          r = r[indices[2]].cast<py::list>();
+          *v = r[indices[3]].cast<int64_t>();
+        });
+        break;
+      case 5:
```

Comment:
we need test for all these cases.

Reply:
Good point, we only test 1~3 dimensional sharding. Let me add a couple of test cases to cover 4th and 5th dimensions, and we should be good.